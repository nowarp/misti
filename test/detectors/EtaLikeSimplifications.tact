contract TestContract {
    // Bad: Example of redundant if statement
    fun redundantIf(condition: Bool): Bool {
        if (condition) {
            return true;
        } else {
            return false;
        }
    }

    // Bad: Example of redundant boolean comparison
    fun redundantComparison(a: Bool): Bool {
        return a == true;
    }

    // Bad: Example of redundant ternary expression
    fun redundantTernary(b: Bool): Bool {
        return b ? true : false;
    }

    // Ok: Correct usage (should not trigger warnings)
    fun correctUsage(condition: Bool): Bool {
        return condition;
    }
}
