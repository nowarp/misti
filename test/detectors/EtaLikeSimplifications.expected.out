[LOW] EtaLikeSimplifications: Simplify 'if' statement by returning the condition directly
test/detectors/EtaLikeSimplifications.tact:4:9:
  3 |     fun redundantIf(condition: Bool): Bool {
> 4 |         if (condition) {
              ^
  5 |             return true;
Help: return condition;
See: https://nowarp.io/tools/misti/docs/detectors/EtaLikeSimplifications

[LOW] EtaLikeSimplifications: Redundant comparison with boolean literal
test/detectors/EtaLikeSimplifications.tact:13:16:
  12 |     fun redundantComparison(a: Bool): Bool {
> 13 |         return a == true;
                      ^
  14 |     }
Help: a
See: https://nowarp.io/tools/misti/docs/detectors/EtaLikeSimplifications

[LOW] EtaLikeSimplifications: Simplify conditional expression by using the condition directly
test/detectors/EtaLikeSimplifications.tact:18:16:
  17 |     fun redundantTernary(b: Bool): Bool {
> 18 |         return b ? true : false;
                      ^
  19 |     }
Help: b
See: https://nowarp.io/tools/misti/docs/detectors/EtaLikeSimplifications